### PR TITLE
StashBuilds.java : try again before asking to "PLEASE SET JENKINS ROOT URL…

### PR DESCRIPTION
**Pull request by jimklimov**
…FROM GLOBAL CONFIGURATION"

https://stackoverflow.com/questions/30355079/jenkins-setting-root-url-via-groovy-api was helpful :)

Note: apparently, https://github.com/jenkinsci/jenkins/commit/c3988aae8ad66fde529830e799fe3d49cd241ebb#diff-1439884ddd002c6ca6463d9657412b15 from Aug 18 2018 broke the earlier working setups, since the constructors doing `load()` were removed with it. Which sort of contradicts some discussion threads from much earlier ages, that the constructor makes a new empty object so a `get()` of existing config should be done. But not being much of a Java developer, I might read it wrong... still, it matches our server's mis-behavior starting a few weeks ago after an update.

### CODE REVIEW

**Comment by jimklimov** [comment]
In site-local testing, this patch did fix our issue. We see links to particular builds in `[BUILD SUCCESS]` parts of the Stash comments again :)

**Comment by jimklimov** [comment]
In site-local testing, this patch did fix our issue. We see links to particular builds in `[BUILD SUCCESS]` parts of the Stash comments again :)

**Comment by jimklimov** [comment]
In the end, the solution recommended with discussion in https://github.com/jenkinsci/jenkins/pull/3570#issuecomment-423195479 was to just access the data using `get()` and avoiding `new`; PR updated accordingly and worked with our Jenkins 2.142 deployment.

**Comment by jimklimov** [comment]
Technically, https://github.com/nemccarthy/stash-pullrequest-builder-plugin/pull/149 is almost identical, but with a null-ness guard seems to be one line better :)

**Review by tomascea — APPROVED** [review]

**Comment by jakub-bochenski** [comment]
> but with a null-ness guard seems to be one line better

@jimklimov it's pointless to do a null check on a `@Nonnull` value

**Comment by jimklimov** [comment]
@jakub-bochenski : Just in case, did you verify since when is the value marked as `@Nonnull`?

My concern here is if the older Jenkins core (as in "compatible-since" metadata of the plugin) already has that, or the double-check is warranted there?

**Comment by jakub-bochenski** [comment]
@jimklimov  good catch, apparently this has only been changed in April https://github.com/jenkinsci/jenkins/commit/79ec930a8ef4f12d2f2574c50c2f42272e5c75fb#diff-1439884ddd002c6ca6463d9657412b15

**Review by matfiej88 — APPROVED** [review]

**Comment by proski** [comment]
Can someone merge it and make a release, please! Even the stable Jenkins branch is at 2.150.1, which requires this fix.

**Comment by jakub-bochenski** [comment]
@proski no, nobody can
Edit: I could swear i posted this link before but apparently I didn't. Sorry, here is the jenkins-dev thread:
https://groups.google.com/forum/m/?utm_medium=email&utm_source=footer#!msg/jenkinsci-dev/oRBRP5-sZe4/9N44G__mDgAJ

Make noise there if you want to make a difference.

**Comment by proski** [comment]
Done (the message is pending approval). Thank you for stepping up!

**Comment by dawidmalina** [comment]
Hopefully soon @jakub-bochenski will be a new project maintainer 👍

**Comment by batmat** [comment]
Processing the request to become maintainer, BTW, I'm thinking: given you apparently submitted many PRs here @jimklimov, would be interested too to become co-maintainer with @jakub-bochenski ?

File: src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java
Comment (timja) [inline src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java:52]: I would recommend removing this comment seems like it adds no value, the reasoning behind the change can be in the commit message or the PR

Comment (jimklimov) [reply src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java:52]: Well, to many newcomers (like myself not so long ago) the code is the documentation. A removed line carries no visibility to `git blame`, not that many people dig in that at all, and a checked-out workspace or a fork has no data from PR discussions. For this particular piece of code, I found too many examples collecting dust on the internet over many years proposing the `new` way to get the config, that without the comment someone might be eager to make this particular mistake again.

**Comment by jakub-bochenski** [comment]
>Processing the request to become maintainer, BTW, I'm thinking: given you apparently submitted many PRs here @jimklimov, would be interested too to become co-maintainer with @jakub-bochenski ?

That would be great. As mentioned I don't plan to actively develop the plugin, only fix the most annoying defects.

**Comment by jimklimov** [comment]
What @jakub-bochenski said :) +1

**Comment by batmat** [comment]
You've both been granted permissions. Thanks again for stepping up!

**Review by jakub-bochenski — APPROVED** [review]
Let's merge it once build has passed

**Comment by jakub-bochenski** [comment]
I guess we need to rebase the branch vs. master for the build to happen

File: src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java
Comment (jakub-bochenski) [inline src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashBuilds.java:54]: We need to add a null check or upgrade to newer jenkins dependency.

My vote is for the former, since the later would reduce compatibility.

**Comment by emangual** [comment]
@jakub-bochenski  Are you guys planning on releasing a new version of the stash-pullrequest-builder-plugin now that this PR has been merged?

**Comment by jakub-bochenski** [comment]
Yes, you can expect a release soon.

@jimklimov no pressure, but could you take a look at #36 #37 #38  ? Those are simple changes, I'd prefer to release with a clean project descriptor.

**Comment by jimklimov** [comment]
Hope I can, but currently preoccupied with other topics on fosdem. So likely next week.

**Pull request merged**
